src/block_loss_functions.rs

```diff
@@ -7,6 +7,7 @@ use crate::graph;
 use crate::model_instance;
 use crate::port_buffer;
 use crate::regressor;
+use log::{error, warn};
```

Comment:
I think it's a bit clearer to actually use `log.error` when logging, rather than importing `log` and then using `error` when logging. It's not super intuitive that these are logging levels

Reply:
Thanks, refactored so that all log calls are done via lib obj itself (log.error does not seem to work here), so
```
log::error!("something");
```
which is more explicit/aligned with what you mention - would that work too? (fddc81a6404136fd16fe0f45e2da2cc41ec594d04)

Reply:
yes, this is what I meant